From 524b8e9d40d82f8df113a3d2cf82f7470457acbd Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Tue, 28 Oct 2025 10:21:38 +0100
Subject: [PATCH 526/553] QXcbAtom: add missing <cstdlib> #include

Clang 21 complained about free() being an undefined identifier.

Amends d67214302f269242ae3d8d2b962fd91ec42c979e.

Pick-to: 6.8 6.5
Change-Id: I16742b06b302576b2ff15fa972d91d1e077f3057
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 6022394cd726e26cb24670f28ecd6eccc99be89a)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/plugins/platforms/xcb/qxcbatom.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/plugins/platforms/xcb/qxcbatom.cpp b/src/plugins/platforms/xcb/qxcbatom.cpp
index 5f712d33d56..e609b84ad5f 100644
--- a/src/plugins/platforms/xcb/qxcbatom.cpp
+++ b/src/plugins/platforms/xcb/qxcbatom.cpp
@@ -9,6 +9,7 @@
 #include <string.h>
 
 #include <algorithm>
+#include <cstdlib>
 
 static const char *xcb_atomnames = {
     // window-manager <-> client protocols
-- 
2.51.2

